### PR TITLE
FIX-#2857: Correctly handle identical index binary operations

### LINKED ISSUE #2857: modin fails to correctly deliver division results in dataframe rows
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Modin version** (`modin.__version__`):  0.9.0
- **Python version**: 3.7.2
- **Code we can use to reproduce**: `new['effratio'] = new['efflen']/new['txtlen']`

### Describe the problem
If you directly use rows in a DataFrame to do division, the result is not correct. I discovered this when I want to calculate a ratio. Some ratio values are larger than 1. I ran the same code with original pandas and it delivered the correct result.

I have listed several items in this dataframe for you to see the problem. See the image.

`new` is a DataFrame object I read in.

issue

### PR DESCRIPTION
**Pull request by devin-petersohn**
Resolves #2857

Signed-off-by: Devin Petersohn <devin.petersohn@gmail.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2857 
- [x] tests added and passing

### CODE REVIEW

File: modin/engines/base/frame/data.py [hunk 1]
```diff
@@ -1783,8 +1783,6 @@ def get_axis_lengths(partitions, axis):
             return self._partitions, [o._partitions for o in other], joined_index
 
         base_frame_idx = non_empty_frames_idx[0]
```
Comment (RehanSD) [inline modin/engines/base/frame/data.py:1785]: Why do we define this variable if we only use it in line 1786? Why not use it in line 1789 as well?

File: modin/engines/base/frame/data.py [hunk 2]
```diff
@@ -973,7 +973,7 @@ def merge(left_index, right_index):
                 return left_index.join(right_index, how=how, sort=sort)
 
         # define condition for joining indexes
-        all_indices_equal = all(indexes[0].equals(index) for index in [indexes[1:]])
+        all_indices_equal = all(indexes[0].equals(index) for index in indexes[1:])
```
Comment (RehanSD) [inline modin/engines/base/frame/data.py:976]: I'm not sure I understand what these changes do - before were we making a 2D list with one row that contains indices, but now we're just looping through the list of indices?

**Review by RehanSD — COMMENTED** [review]
Left some comments for clarification.

File: modin/engines/base/frame/data.py (hunk 2 above)
Comment (devin-petersohn) [reply modin/engines/base/frame/data.py:976]: Yes, it was a bug. Deleting these two characters fixes the case in the issue.

File: modin/engines/base/frame/data.py (hunk 1 above)
Comment (devin-petersohn) [reply modin/engines/base/frame/data.py:1785]: The code quality in this helper method is in need of some TLC, I do not disagree. Let's put that later.

**Review by RehanSD — APPROVED** [review]
Looks good to me!

**Pull request merged**